streams/src/main/java/org/apache/kafka/streams/state/internals/InMemoryTimeOrderedKeyValueBuffer.java

```diff
@@ -57,6 +60,8 @@
     private long minTimestamp = Long.MAX_VALUE;
     private RecordCollector collector;
     private String changelogTopic;
+    private Sensor bufferSizeSensor;
```

Comment:
In the wiki there is another metric for `suppression-mem-buffer-evict`, is that not in the scope of this PR?

Reply:
Ah, now that you mention it, when I originally created this PR, I noticed there were some shortcomings with the metrics I proposed in the KIP, so I wanted to get the reviewers' feedback on the form it has taken in this PR before proposing an update to the KIP.

In particular, I proposed an asymmetric pair of metrics where the processor node would measure the number of incoming events (`intermediate-result-suppression`), but the buffer would measure the outgoing (emitted==evicted) events. It seemed better to offer a symmetric pair of in/out metrics on either the processor or the buffer. In this PR, I chose to put them both on the node.

In other words, I'm proposing to replace `suppression-mem-buffer-evict` with `suppression-emit`.

Of course, in this same review, you have proposed to drop `intermediate-result-suppression` in favor of the existing `process-rate` and `process-total` node-level metrics. This reduces the argument, but I *think* it still makes sense to make this a node metric instead of a buffer metric, since (IMHO) it more closely matches the intent of using a suppression *node* to control the emission rate.

Reply:
Thanks for the clarification, that sounds good to me.